Core.Auditing/IfxAuditLogger.cs

```diff
@@ -1,101 +1,124 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 
-using Microsoft.Cloud.InstrumentationFramework;
 using Microsoft.CloudMine.Core.Telemetry;
+using Microsoft.Extensions.Logging;
+using OpenTelemetry.Audit.Geneva;
 using System;
 using System.Collections.Generic;
 using System.Net;
 using System.Net.Sockets;
 
 namespace Microsoft.CloudMine.Core.Auditing
 {
+
+    public class TargetResource
+    {
+        public string type;
+        public string name;
+        public string cluster;
+        public string region;
+
+        public TargetResource(string type, string name, string cluster = null, string region = null)
+        {
+            this.type = type;
+            this.name = name;
+            this.cluster = cluster;
+            this.region = region;
+        }
+    }
+
+    public class CallerIdentity
```

Comment:
>CallerIdentity [](http://example.com/codeflow?start=17&length=14)

Interesting, so open telemetry no longer provides these classes in the package?

Reply:
Added

Reply:
Correct, I do not understand why this is the case. They provide classes `TargetResourceEntity` and `CallerIdentityEntity` but they are not designed for use like this.

Reply:
What about these classes https://github.com/microsoft/opentelemetry-dotnet-exporters/blob/main/src/OpenTelemetry.Audit.Geneva/DataModel/CallerIdentityEntry.cs?

Reply:
Yes these are the classes I mentioned above. The problem with them is that they do not include a field for `CallerIdentityType` and` `TargetResourceType`  which is data we need here. Additionally there is no method in `AuditLog` to add these objects, only methods to pass in the field values + type as strings.

Reply:
Sounds good. Nitpick: can we capitalize the properties? Thanks

Reply:
+1 on proper naming conventions. Also, if these are properties, they need proper accessors. If not, they should be private (fields).